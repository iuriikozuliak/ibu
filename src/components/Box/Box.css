@value success, dark from 'theme/colors.css';
@value unit, unit2x, unit4x from 'theme/size.css';

.Box {
  composes: box from 'theme/layout.css';
}

.Box.Bordered {
  border: 1px solid dark;
}

.Box.Big {
  text-align: center;
  padding: unit4x;
  margin: unit2x 0 0;
}

.Box.Success {
  border-color: success;
}

.Box.Rounded {
  border-radius: unit
}
