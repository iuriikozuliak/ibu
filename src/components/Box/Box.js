import React from 'react';

import styles from './Box.css';

const Box = ({ children }) => (
  <div className = { styles.Box }>
    { children }
  </div>
);

export default Box;
