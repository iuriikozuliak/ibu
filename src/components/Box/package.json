{
  "name": "Box",
  "private": true,
  "main": "./Box.js"
}
