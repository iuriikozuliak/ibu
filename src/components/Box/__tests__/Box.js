import React    from 'react';
import renderer from 'react-test-renderer';
import Box      from '../Box';

test('Renders Box component', () => {
  const component = renderer.create(
    <Box />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Renders Box component with passed children', () => {
  const component = renderer.create(
    <Box>
      <p>Hi!</p>
    </Box>
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Renders Box component with passed children and isBordered modifier', () => {
  const component = renderer.create(
    <Box isBordered>
      <p>Hi!</p>
    </Box>
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Renders Box component with passed children and isBig modifier', () => {
  const component = renderer.create(
    <Box isBig>
      <p>Hi!</p>
    </Box>
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Renders Box component with passed children and isSuccess modifier', () => {
  const component = renderer.create(
    <Box isSuccess>
      <p>Hi!</p>
    </Box>
  ).toJSON();
  expect(component).toMatchSnapshot();
});

