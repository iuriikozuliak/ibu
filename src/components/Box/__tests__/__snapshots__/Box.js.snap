// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Box component 1`] = `
<div
  className="Box"
/>
`;

exports[`Renders Box component with passed children 1`] = `
<div
  className="Box"
>
  <p>
    Hi!
  </p>
</div>
`;

exports[`Renders Box component with passed children and isBig modifier 1`] = `
<div
  className="Box Big"
>
  <p>
    Hi!
  </p>
</div>
`;

exports[`Renders Box component with passed children and isBordered modifier 1`] = `
<div
  className="Box Bordered"
>
  <p>
    Hi!
  </p>
</div>
`;

exports[`Renders Box component with passed children and isSuccess modifier 1`] = `
<div
  className="Box Success"
>
  <p>
    Hi!
  </p>
</div>
`;
